### DISCUSSION TITLE
Welcome to django-htmx-patterns Discussions!

### DISCUSSION THREAD

**Discussion by spookylukey** [spookylukey/django-htmx-patterns#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to share ideas and feedback. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded.

**Comment by ccll** [comment]
Big thanks for the articles, this is just the thing I'm looking for, for a long time.

I'm interested in Django+HTMX a long time ago, but the idea of managing two template files for 'full page render' versus 'partial render' and check the request header and switch template in the view is a big NO to me.

And your clever use of 'django-render-block' with decorators totally nail it.
I'm using your method in my next project.

**Comment by cristianrat** [comment]
I have a question. In your examples I see you're using very few packages.
Why not have any of the django-components, or template-partials, cotton etc.?
I'm in a bit of a pickle, not sure if I should use any of them (have used django-components for some pages, but am not sure I really needed it :) )

**Reply by killianarts** [reply]
Cotton is amazing and works well with HTMX. I have a form component that looks like this:

```html
<c-vars method="post" dj-block="content" hx-target="#dashboard-content" hx-swap="outerHTML" url="." />
<div class="form-dj-block-designator" hx-vals='{"use_block": "{{ dj_block }}"}'>
  <form action="{{ url }}" method="{{ method }}" hx-{{ method }}="{{ url }}" hx-swap="{{ hx_swap }}" hx-target="{{ hx_target }}" {{ attrs }}>
    {% csrf_token %}
      {{ slot }}
  </form>
</div>
```

With this form, I have `use_block` set by default to my `content` block. `use_block` is set on a div over the form so that I can override it with a button if I want to.

I then follow the `@for_htmx` decorator pattern.

**Reply by spookylukey** [reply]
Yep, I deliberately used the minimum of 3rd party packages, mostly because I want to talk about the fundamental htmx + django patterns.

The other packages you choose depend on other needs and preferences. For example, component libraries like django-components and cotton depend on how much you go down the "component" line, which depends on how big your project is, and I think it also ends up depending on your approach to CSS, like the big "Tailwind or not". (If you use a more classic CSS approach like [CubeCSS](https://cube.fyi/), then your "heading components" are `<h1>`, `<h2>` etc. but if you use Tailwind you're probably going to want a component just to do headings)

But people have a lot of different opinions on this, so the docs in this repo choose to go with the minimum. My recommendation would be to add new packages when you have specific reasons and needs to add them, know exactly why you add them, and [be sure that they are worth the complexity you are adding](https://lukeplant.me.uk/blog/posts/no-one-actually-wants-simplicity/).


I haven't tried to use django-template-partials, it overlaps a lot with the method I use, but I haven't compared the patterns in detail.

**Comment by cristianrat** [comment]
Makes sense :)
Thank you.
Will try to stick with no packages for some new apps I am adding to my
django project
Thanks for making this repo public, it's helpful :)
